cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -199,6 +200,8 @@ protected void configure() {
           bind(MetricDatasetFactory.class).to(DefaultMetricDatasetFactory.class).in(Scopes.SINGLETON);
           bind(MetricStore.class).to(DefaultMetricStore.class);
           bind(DatasetFramework.class).to(InMemoryDatasetFramework.class).in(Scopes.SINGLETON);
+          // Upgrade tool does not need to record lineage for now.
+          bind(LineageWriter.class).to(NoOpLineageWriter.class);
```

Comment:
Another thing to think about is if we should pull in all the guice bindings for metadata into metadata service module. Not for this PR, but maybe a Jira